packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -113,24 +180,62 @@ function updateStateFromFacetResponse(
   facetRequest.numberOfValues = facetResponse.values.length;
   facetRequest.field = facetResponse.field;
   facetRequest.type = facetResponse.type;
-  facetRequest.values = (facetResponse as FacetResponse).values.map(
-    convertFacetValueToRequest
-  );
+  switch (facetResponse.type) {
+    case 'regular':
+      facetRequest.values = (facetResponse as RegularFacetResponse).values.map(
+        convertFacetValueToRequest
+      );
+      break;
+    case 'numericalRange':
+      facetRequest.values = convertToRangeRequests(
+        (facetResponse as NumericFacetResponse).values
+      );
+      break;
+    case 'dateRange':
+      // TODO
+      break;
+    case 'hierarchical':
+      // TODO
+      break;
+  }
+
   facetRequest.preventAutoSelect = false;
 }
 
 function insertNewValue(
   facetRequest: CommerceFacetRequest,
-  facetValue: FacetValueRequest
+  facetValue: FacetValueRequest | NumericRangeRequest
 ) {
-  const {values} = facetRequest;
+  const {type, values} = facetRequest;
   const firstIdleIndex = values.findIndex((v) => v.state === 'idle');
   const indexToInsertAt =
     firstIdleIndex === -1 ? values.length : firstIdleIndex;
 
   const valuesBefore = values.slice(0, indexToInsertAt);
   const valuesAfter = values.slice(indexToInsertAt + 1);
 
-  facetRequest.values = [...valuesBefore, facetValue, ...valuesAfter];
+  switch (type) {
+    case 'regular':
+      facetRequest.values = [
+        ...(valuesBefore as FacetValueRequest[]),
+        facetValue as FacetValueRequest,
+        ...(valuesAfter as RegularFacetValue[]),
+      ];
+      break;
+    case 'numericalRange':
+      facetRequest.values = [
+        ...(valuesBefore as NumericRangeRequest[]),
+        facetValue as NumericRangeRequest,
+        ...(valuesAfter as NumericRangeRequest[]),
+      ];
+      break;
+    case 'dateRange':
+      // TODO
+      break;
+    case 'hierarchical':
+      // TODO
+      break;
+  }
```

Comment:
If this switch job is to define the values, to simplify the code and make it easier to read, might be good to put it in another func